week4/Q1.py

```diff
@@ -0,0 +1,104 @@
+def _rename_islands(islands_map, old_name, new_name):
+	'''
+	Changes names of the tiles.
+
+	Args:
+		islands_map: two-dimensional list
+		old_name: int
+		new_name: int
+
+	Returns:
+		two-dimensional list
+	'''
+	for i, row in enumerate(islands_map):
+		for j, tile in enumerate(row):
+			if tile == old_name:
+				islands_map[i][j] = new_name
+	return islands_map
+
+
+def _name_island(islands_map, row, column, islands, island_counter, upper_tile, left_tile):
+	'''
+	Gives name (which island) to the new tile, renames old tiles if needed (if islands which were seen as separate actually happen to be connected).
```

Comment:
This comment is not clear: I cannot understand by reading this what the function is meant to do.
It would be nice to rephrase it to be clearer and make sure you refer to things by the right name.

For example, you talk about an island name, but I think that refers to the `island_counter`.

Moreover, make sure you use clear names for things.

For example, `island_counter` would seem to contain the number of islands, but you also use
the counter as the name of an island. It is okay to do so, but it would be nicer if you can abstract
that, so that counters and names are not mixed in the code.